lib/vault.py

```diff
@@ -210,6 +210,28 @@ def configure(self):
                 client.write('/auth/aws-ec2/role/' + policy, policies = policy,
                                                              bound_iam_role_arn = arn_prefix + policy)
 
+        # AWS Authentication Backend
+        if aws_creds is None:
+            print("Vault AWS credentials files does not exist, skipping configuration of AWS-EC2 authentication backend")
+        else:
+            try:
+                #Enable AWS EC2 auth in Vault
+                client.enable_auth_backend('aws')
+            except hvac.exceptions.InvalidRequest as ex:
+                print("aws auth backend already created.")
+            
+            #Configure the credentials required to make AWS API calls
+            client.write('auth/aws/config/client', access_key = aws_creds["aws_access_key"],
+                                                   secret_key = aws_creds["aws_secret_key"])
+            #Define policies and arn                                     
+            policies = [p for p in provisioner_policies if p not in ('provisioner',)]
+            arn = 'arn:aws:iam::{}}:instance-profile/'.format(aws_creds["aws_account"])
```

Comment:
Need a temporary way to pass the AWS account id around. Eventually the refactored BOSS config file can be used.